fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/qrcode/CameraPermissionFragment.java

```diff
@@ -6,87 +6,75 @@
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.Button;
 
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.ActivityResultRegistry;
+import androidx.activity.result.contract.ActivityResultContracts;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.core.app.ActivityCompat;
 import androidx.fragment.app.Fragment;
-import androidx.fragment.app.FragmentActivity;
 
-import com.github.dedis.popstellar.R;
 import com.github.dedis.popstellar.databinding.QrcodeCameraPermFragmentBinding;
-import com.github.dedis.popstellar.ui.detail.LaoDetailActivity;
-import com.github.dedis.popstellar.ui.detail.LaoDetailViewModel;
-import com.github.dedis.popstellar.ui.home.HomeActivity;
 
 /** Fragment handling permission granting for the camera */
 public final class CameraPermissionFragment extends Fragment {
 
   public static final String TAG = CameraPermissionFragment.class.getSimpleName();
-
-  private static final int HANDLE_CAMERA_PERM = 2;
+  public static final String REQUEST_KEY = "PERMISSION_REQUEST";
 
   private QrcodeCameraPermFragmentBinding mCameraPermFragBinding;
-  private CameraPermissionViewModel mCameraPermissionViewModel;
+  private final ActivityResultLauncher<String> requestPermissionLauncher;
+
+  public CameraPermissionFragment(@NonNull ActivityResultRegistry resultRegistry) {
```

Comment:
That's what I found in the android documentation.

But also, the bundle is used in the other lifecycle methods. Not the creation of the instance.
I think that the instance is kept somewhere because otherwise, we would only provide a class of the fragment, not the instance.